serving/src/test/java/ai/djl/serving/http/ClusterRequestHandlerTest.java

```diff
@@ -0,0 +1,263 @@
+ * OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+package ai.djl.serving.http;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyLong;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.mockito.MockitoAnnotations.openMocks;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+
+import ai.djl.serving.Arguments;
+import ai.djl.serving.util.ConfigManager;
+
+import io.netty.channel.embedded.EmbeddedChannel;
+import io.netty.handler.codec.http.DefaultFullHttpRequest;
+import io.netty.handler.codec.http.DefaultFullHttpResponse;
+import io.netty.handler.codec.http.HttpMethod;
+import io.netty.handler.codec.http.HttpVersion;
+
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.io.FileUtils;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.mockito.invocation.InvocationOnMock;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+
+public class ClusterRequestHandlerTest {
+    private static final String TEST_DIRECTORY = "/tmp/test-clusterrequesthandler";
+
+    @Mock private ProcessBuilder mockProcessBuilder;
+    @Mock private Process sshProcess;
+    private String expectedKey;
+    private String fileToDelete;
+    private String originalHome;
+
+    private ClusterRequestHandler handler;
+
+    @BeforeClass
+    public void setup() throws Exception {
+        originalHome = System.getProperty("user.home");
+        System.setProperty("user.home", TEST_DIRECTORY);
+        Files.createDirectories(Path.of(TEST_DIRECTORY + "/.ssh"));
+
+        ConfigManager.init(new Arguments(CommandLine.builder().build()));
+    }
+
+    @BeforeMethod
+    public void setupTest() throws Exception {
+        openMocks(this).close();
+        handler = new ClusterRequestHandler(cmd -> mockProcessBuilder.command(cmd));
+        setupMockProcessBuilder();
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void tearDownClass() throws Exception {
+        System.setProperty("user.home", originalHome);
+        FileUtils.deleteDirectory(new File(TEST_DIRECTORY));
```

Comment:
We can use `ai.djl.utils.Utils.deleteQuietly(<path>)`. We typically use that in our tests to recursively delete a directory and all it's contents (it does deletion without throwing any exceptions)

Reply:
Let me make it configurable with this CR. I did not like updating System properties in the test

Reply:
One thing is that if the path is changed after the handler is initialized, it would create new ssh key on the next invocation. Do we want to allow that? I can also limit this change to be a protected method, to be configurable with test only